### PR TITLE
python3Packages.jsonargparse: cleanup, skip failing tests

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 497970`
Commit: `2f5a948513aac2b0c2c396ec83290e686b6ca586`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 9 packages marked as broken and skipped:</summary>
  <ul>
    <li>local-ai</li>
    <li>python313Packages.pyannote-audio</li>
    <li>python313Packages.pyannote-audio.dist</li>
    <li>python313Packages.whisperx</li>
    <li>python313Packages.whisperx.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>whisperx</li>
    <li>whisperx.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 38 packages built:</summary>
  <ul>
    <li>calibre</li>
    <li>convertx</li>
    <li>nixpkgs-manual</li>
    <li>piper-tts</li>
    <li>piper-tts.dist</li>
    <li>python313Packages.fastmri</li>
    <li>python313Packages.fastmri.dist</li>
    <li>python313Packages.finetuning-scheduler</li>
    <li>python313Packages.finetuning-scheduler.dist</li>
    <li>python313Packages.gluonts</li>
    <li>python313Packages.gluonts.dist</li>
    <li>python313Packages.jsonargparse</li>
    <li>python313Packages.jsonargparse.dist</li>
    <li>python313Packages.lightning</li>
    <li>python313Packages.lightning-utilities</li>
    <li>python313Packages.lightning-utilities.dist</li>
    <li>python313Packages.lightning.dist</li>
    <li>python313Packages.pytorch-lightning</li>
    <li>python313Packages.pytorch-lightning.dist</li>
    <li>python313Packages.torchmetrics</li>
    <li>python313Packages.torchmetrics.dist</li>
    <li>python314Packages.fastmri</li>
    <li>python314Packages.fastmri.dist</li>
    <li>python314Packages.finetuning-scheduler</li>
    <li>python314Packages.finetuning-scheduler.dist</li>
    <li>python314Packages.jsonargparse</li>
    <li>python314Packages.jsonargparse.dist</li>
    <li>python314Packages.lightning</li>
    <li>python314Packages.lightning-utilities</li>
    <li>python314Packages.lightning-utilities.dist</li>
    <li>python314Packages.lightning.dist</li>
    <li>python314Packages.pytorch-lightning</li>
    <li>python314Packages.pytorch-lightning.dist</li>
    <li>python314Packages.torchmetrics</li>
    <li>python314Packages.torchmetrics.dist</li>
    <li>unbook</li>
    <li>wyoming-piper</li>
    <li>wyoming-piper.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/jsonargparse/default.nix [hunk 1]
```diff
@@ -36,18 +46,18 @@ buildPythonPackage (finalAttrs: {
...
-      typeshed-client
-      requests
-    ];
+  optional-dependencies = lib.fix (self: {
+    all =
+      self.argcomplete
+      ++ self.fsspec
+      ++ self.jsonnet
+      ++ self.jsonschema
+      ++ lib.omegaconf
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/development/python-modules/jsonargparse/default.nix]: Fixed!

**Review by kirillrdy — APPROVED** [review]
